src/pudl/transform/eia860.py

```diff
@@ -1021,3 +1021,88 @@ def _core_eia860__boiler_stack_flue(
     )
 
     return bsf_assn
+
+
+@asset
+def _core_eia860__cooling_equipment(
+    raw_eia860__cooling_equipment: pd.DataFrame,
+) -> pd.DataFrame:
+    """Transform the EIA 860 cooling equipment table.
+
+    - spot clean year values before converting to dates
+    - standardize water rate units to gallons per minute (2009-2013 used cubic
+      feet per second)
+    - convert kilodollars to normal dollars
+    """
+    ce_df = raw_eia860__cooling_equipment
+
+    # Generic cleaning
+    ce_df = ce_df.pipe(pudl.helpers.fix_eia_na).pipe(pudl.helpers.add_fips_ids)
+
+    # Spot cleaning and date conversion
+    ce_df.loc[
+        ce_df["chlorine_equipment_operating_year"] == 971,
+        "chlorine_equipment_operating_year",
+    ] = 1971
+    ce_df.loc[ce_df["pond_operating_year"] == 0, "pond_operating_year"] = np.nan
+    ce_df.loc[ce_df["tower_operating_year"] == 974, "tower_operating_year"] = 1974
+    ce_df = ce_df.pipe(pudl.helpers.month_year_to_date).pipe(
+        pudl.helpers.convert_to_date
+    )
+    # Convert cubic feet/second to gallons/minute
+    one_cfs_in_gpm = 448.8311688
+    conversion_cols = {
+        "tower_water_rate_100pct_cubic_feet_per_second": "intake_rate_100pct_gallons_per_minute",
+        "intake_rate_100pct_cubic_feet_per_second": "tower_water_rate_100pct_gallons_per_minute",
+    }
+    for col_seconds, col_minutes in conversion_cols.items():
+        ce_df.loc[ce_df["report_date"].dt.year.isin(range(2009, 2013)), col_minutes] = (
+            ce_df[col_seconds] * one_cfs_in_gpm
+        )
+
+    # Convert thousands of dollars to dollars and remove suffix from column name
+    ce_df.loc[:, ce_df.columns.str.endswith("_thousand_dollars")] *= 1000
+    ce_df.columns = ce_df.columns.str.replace("_thousand_dollars", "")
+
+    return ce_df
+
+
+@asset_check(asset=_core_eia860__cooling_equipment, blocking=True)
+def cooling_equipment_check(cooling_equipment):
+    """Check data quality.
+
+    - only completely null cols are tower type 3 and 4
+    - no measurement cols move by 80% or more over the course of one report period.
+    """
+    expected_null_cols = {"tower_type_3", "tower_type_4"}
+    pudl.validate.no_null_cols(
+        cooling_equipment,
+        cols=set(cooling_equipment.columns) - expected_null_cols,
+    )
+    smoothly_varying_cols = [
```

Comment:
Just took a stab at "stuff that seems like it should mostly move smoothly." We could easily extract this into a utility function, and even include different thresholds per column - but figured this was OK for a quick & dirty check.

Reply:
This makes sense intuitively but I'm wary of defining completely ad hoc validation tests for each table, ideally we'd have uniform validation expectations across each dataset at least.